### PR TITLE
refactor(reportBox): translate to functional component with hooks

### PR DESCRIPTION
**Pull request by cbernat**
Refactor ReportBox from class component to functional component with hooks.
Issue: DW-93

### CODE REVIEW

File: src/components/Reports/ReportsBox/ReportsBox.js [hunk 1]
```diff
@@ -1,122 +1,77 @@
...
-    super();
+/**
+ * @param { Object } props - props
+ * @param { import('../../../services/pure-di').AppServices } props.dependencies
+ */
+const ReportsBox = ({
+  domainName,
+  dateFrom,
+  dateTo,
+  withEmail,
```
Comment (fcoronelmakingsense) [inline src/components/Reports/ReportsBox/ReportsBox.js]: maybe can use a constant `emailFilter` with the values 'with_email' 'without_email or null

File: src/components/Reports/ReportsBox/ReportsBox.js [hunk 2]
```diff
@@ -1,122 +1,77 @@
...
+}) => {
+  const [visits, setVisits] = useState(null);
 
-    this.datahubClient = datahubClient;
-
-    this.state = {
-      visits: null,
+  useEffect(() => {
+    const fetchVisitsByPeriod = async (domainName, dateFrom, dateTo) => {
+      let asyncRequest = datahubClient.getTotalVisitsOfPeriod({
```
Comment (fcoronelmakingsense) [inline src/components/Reports/ReportsBox/ReportsBox.js]: what do you think to use `const asyncRequest =  await datahubClient.getTotalVisitsOfPeriod({` ? no need to put in null after

File: src/components/Reports/ReportsBox/ReportsBox.js [hunk 3]
```diff
@@ -1,122 +1,77 @@
...
-class ReportsBox extends React.Component {
-  /**
-   * @param { Object } props - props
-   * @param { import('../../../services/pure-di').AppServices } props.dependencies
-   */
-  constructor({ dependencies: { datahubClient } }) {
-    super();
+/**
+ * @param { Object } props - props
+ * @param { import('../../../services/pure-di').AppServices } props.dependencies
```
Comment (andresmoschini) [inline src/components/Reports/ReportsBox/ReportsBox.js]: These annotations are wrong, right?

Comment (andresmoschini) [reply src/components/Reports/ReportsBox/ReportsBox.js]: Oh, no, I was reading it wrong, sorry. Deleting the comments...

**Comment by cbernat** [comment]
@fcoronelmakingsense @andresmoschini if there are no further issues, I will rebase and merge this.

**Pull request merged**